cpp/src/arrow/util/bit_util.h

```diff
@@ -118,6 +118,16 @@ constexpr uint64_t LeastSignificantBitMask(int64_t bit_index) {
   return (static_cast<uint64_t>(1) << bit_index) - 1;
 }
 
+// Returns a mask for the bit_index lower order bits.
+// Only valid for bit_index in the range [0, sizeof(Uint)].
+template <typename Uint>
+constexpr auto LeastSignificantBitMaskInc(Uint bit_index) {
```

Comment:
What does `Inc` mean here?

And I wonder if we cannot just reconcile the two functions, for example:
```c++
template <typename Uint, bool kAllowUpperBound = false>
constexpr auto LeastSignificantBitMask(Uint bit_index) {
  if constexpr (kAllowUpperBound) {
    if (bit_index == 8 * sizeof(Uint)) {
      return ~Uint{0};
    }
  }
  return (Uint{1} << bit_index) - Uint{1};
}
```

Reply:
`Inc` was for inclusive but your solution is good too!